src/terrain/util/keycloak_oidc.clj

```diff
@@ -29,14 +29,36 @@
       (throw+ (ex-info (str "Missing required JWT claims: " missing)
                        {:type :validation :cause :missing-fields})))))
 
+(def cached-certs (atom nil))
+(def cached-certs-time (atom nil))
+(def cache-ttl (* 24 60 60 1000)) ;; 1 day, milliseconds
+
+(defn- update-cert-cache
+  []
+  (otel/with-span [s ["update-cert-cache"]]
+    (let [new-certs (kc/get-oidc-certs)]
+      (reset! cached-certs-time (System/currentTimeMillis))
+      (reset! cached-certs new-certs))))
+
+(defn- validate-with-cache
```

Comment:
in short, here, theoretically:

check if there's a cached value and it's not expired
if there is, try to validate with it
if either thing above failed, validate again with an updated cache, and keep/return the error this time if there is one